File: pi_euler/pi.go
// This program give an approach to pi using the solution of Euler to the famous series 1/x²

package main

import (
    "fmt"
    "math"
)

var sum, term, epsilon, i, pi  float64

func main() {
    epsilon = 1E-16
    sum = 0
    term = 1
    for term > epsilon {
    i += 1
    term = 1/(i*i)
    sum += term
    }

    pi =  math.Sqrt(sum*6)
    fmt.Printf("The approach of pi using an epsilon of %v is %v\n",epsilon, pi)
}

File: pi_euler/pi.rs
// This program give an approach to pi using the solution of Euler to the famous series 1/x²

fn main(){
    let mut sum: f64 = 0.0;
    let epsilon: f64 = 1E-16;
    let pi: f64;
    let mut term: f64;
    let mut i:f64 = 0.0;
    loop {
        i += 1.0;
        term = 1.0/(i*i);
        sum += term;
        if term < epsilon {
            break;
        }
        //println!("{} {} {}", term, epsilon,i);
    }
    pi = sum*6.0;
    println!("The approach of pi using an epsilon of {} is {}", epsilon, pi.sqrt());
}
